src/renderer/redux/selectors/discover.js

```diff
@@ -0,0 +1,35 @@
+import { createSelector } from 'reselect';
+import { selectFeaturedUris } from './content';
+import { selectClaimsById, selectAllClaimsByChannel } from './claims';
+
+export const selectDiscover = createSelector(
+  [selectFeaturedUris, selectClaimsById, selectAllClaimsByChannel],
+  (featuredUris, claimsById, claimsByChannel) => {
+    let categories = featuredUris;
+    if (!!categories && !!claimsByChannel) {
+      let channels = [];
+      Object.keys(categories).forEach(key => {
+        if (key.indexOf('@') === 0) {
+          channels.push(key);
+        }
+      });
+      Object.keys(claimsByChannel).forEach(key => {
+        if (channels.includes(key)) {
+          delete categories[key];
+          const ids = claimsByChannel[key][1];
+          let uris = [];
+          const newKey = `${key}#${claimsById[ids[0]].value.publisherSignature.certificateId}`;
+          ids.forEach(id => {
```

Comment:
instead of all of these loops, you should learn how to use `map`, `reduce`, `filter`, etc.

i dont think this would block merging, however, here is a rough sketch of how i would do this

```
uri = ids.filter(id => claimsById[id]).map(id => `${claimsById[id.name}#${claimsById[id.claim_id}`);
```

and in fact you could even set

```
categories[newKey] = ids.filter(id => claimsById[id]).map(id => `${claimsById[id.name}#${claimsById[id.claim_id}`);
```

this could cut out a lot of this code and make it shorter. It's something you should play around with.